### PR TITLE
Helm Charts creation

### PR DESCRIPTION
**Pull request by anoop1402**
## Summary and Scope

_Summarize what has changed. Explain why this PR is necessary. What is impacted? Is this a new feature, critical bug fix, etc?_

_Is this change backwards incompatible, backwards compatible, or a backwards compatible bugfix?_

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [issue id](issue link)
* Change will also be needed in `<insert branch name here>`
* Future work required by [issue id](issue link)
* Documentation changes required in [issue id](issue link)
* Merge with/before/after `<insert PR URL here>`

## Testing

_List the environments in which these changes were tested._

### Tested on:

  * `<development system>`
  * Local development environment
  * Virtual Shasta

### Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? If not, why?
- Was downgrade tested? If not, why?
- Were new tests (or test issues/Jiras) created for this change?

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_

## Pull Request Checklist

- [ ] Target branch correct
- [ ] Testing is appropriate and complete, if applicable
- [ ] Is a new version being released? Update https://github.com/Cray-HPE/cray-product-catalog/wiki/CSM-Compatibility-Matrix
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Review by lathanm — APPROVED** [review]

File: .gitignore
Comment (arnab-muk-hpe) [inline .gitignore:1]: We need to keep the actual.

File: charts/migration/templates/job.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    spec:
+      containers:
+        - args:
+          - -c
+          - /usr/bin/catalog_migrate
+          command:
+          - /bin/ash
+          env:
+          - name: PRODUCT
+            value: {{ .Values.job.product }}
```
Comment (arnab-muk-hpe) [inline charts/migration/templates/job.yaml:44]: Not required

File: charts/migration/templates/job.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          - name: PRODUCT_VERSION
```
Comment (arnab-muk-hpe) [inline charts/migration/templates/job.yaml:45]: Not required

File: charts/migration/templates/job.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            value: {{ .Values.job.productVersion }}
+          - name: CONFIG_MAP
+            value: {{ .Values.job.configMap }}
+          - name: CONFIG_MAP_NAMESPACE
+            value: {{ .Values.job.configMapNamespace }}
+          - name: YAML_CONTENT
```
Comment (arnab-muk-hpe) [inline charts/migration/templates/job.yaml:51]: Not required

Comment (arnab-muk-hpe) [reply charts/migration/templates/job.yaml:51]: Only config map and namespace required.

File: charts/migration/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+#
+# MIT License
+#
+# (C) Copyright 2021-2023 Hewlett Packard Enterprise Development LP
```
Comment (krabhi-hpe) [inline charts/migration/Chart.yaml:4]:
```suggestion
# (C) Copyright 2023 Hewlett Packard Enterprise Development LP
```

File: charts/migration/templates/job.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+spec:
+  backoffLimit: 0
+  template:
+    spec:
+      containers:
+        - args:
+          - -c
+          - /usr/bin/catalog_migrate
+          command:
+          - /bin/ash
```
Comment (arnab-muk-hpe) [inline charts/migration/templates/job.yaml:41]: it has to be /bin/bash or  /bin/sh

**Review by arnab-muk-hpe — CHANGES_REQUESTED** [review]
Anoop run the helms from branch once.
Added few changes.

File: charts/migration/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+apiVersion: v2
+name: cpc-migration
+version: 0.0.0-chart  # Replaced during build
+description: Helm Chart for splitting the main cm
+keywords:
+  - cray
+  - shasta
+sources:
```
Comment (krabhi-hpe) [inline charts/migration/Chart.yaml:31]: @anoop1402 : What is `sources` used for?

File: charts/migration/templates/job.yaml [hunk 7]
```diff
@@ -0,0 +1,61 @@
+{{/*
+MIT License
+
+(C) Copyright 2021-2023 Hewlett Packard Enterprise Development LP
```
Comment (krabhi-hpe) [inline charts/migration/templates/job.yaml:4]:
```suggestion
(C) Copyright 2023 Hewlett Packard Enterprise Development LP
```

File: charts/migration/templates/job.yaml [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+            value: {{ .Values.job.configMap }}
+          - name: CONFIG_MAP_NAMESPACE
+            value: {{ .Values.job.configMapNamespace }}
+          - name: YAML_CONTENT
+            value: {{ .Values.job.yamlContent }}
+          securityContext:
+            runAsUser: 0
+          image: {{ .Values.job.image }}
+          imagePullPolicy: IfNotPresent
+          name: update-catalog
```
Comment (krabhi-hpe) [inline charts/migration/templates/job.yaml:57]:
```suggestion
          name: migrate-catalog
```

File: charts/migration/values.yaml [hunk 9]
```diff
@@ -0,0 +1,39 @@
+#
+# MIT License
+#
+# (C) Copyright 2021-2023 Hewlett Packard Enterprise Development LP
```
Comment (krabhi-hpe) [inline charts/migration/values.yaml:4]:
```suggestion
# (C) Copyright 2023 Hewlett Packard Enterprise Development LP
```

File: charts/migration/values.yaml [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+job:
+  image: cpc-image:latest
+  image1: bitnami/kubectl:latest
+  product: csm
+  productVersion: 1.16.17
```
Comment (krabhi-hpe) [inline charts/migration/values.yaml:29]: What does this specific version mean ? Is it like only on this version of product this job will be run?

**Review by krabhi-hpe — CHANGES_REQUESTED** [review]

File: charts/migration/Chart.yaml [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+apiVersion: v2
+name: cpc-migration
+version: 0.0.0-chart  # Replaced during build
+description: Helm Chart for splitting the main cm
```
Comment (krabhi-hpe) [inline charts/migration/Chart.yaml:27]:
```suggestion
description: Helm Chart for splitting the 'cray-product-catlog' configMap
```

**Pull request closed without merging**